make doctest fails locally

I've been working on #93 and I'm having trouble running the tests locally.

I'm still pretty new to Haskell development, and don't have prior experience working with cabal projects that don't use the stack build tool.

In order to get this project to build locally, I initially ran `make`, which failed because I didn't have cabal installed.

Eventually, I got it to build by installing cabal through stack (since I already had stack installed).
`stack install cabal-install`

Now when I run `make`, the project is compiled.

However, if I run `make doctest` (after having previously installed doctest via `stack install doctest`), doctest seems unable to find some imports:

```
➜  recursion-schemes git:(master) make doctest                                                                                                         
doctest -DCURRENT_PACKAGE_KEY='"recursion-schemes"' -Iinclude src                                                                                      
                                                                                                                                                       
src/Data/Functor/Foldable.hs:93:1: error:                                                                                                              
    Could not find module ‘Control.Comonad’                                                                                                            
    Perhaps you meant Control.Monad (from base-4.12.0.0)                                                                                               
    Use -v to see a list of the files searched for.                                                                                                    
   |                                                                                                                                                   
93 | import Control.Comonad                                                                                                                            
   | ^^^^^^^^^^^^^^^^^^^^^^                                                                                                                            
                                                                                                                                                       
src/Data/Functor/Foldable.hs:94:1: error:                                                                                                              
    Could not find module ‘Control.Comonad.Trans.Class’                                                                                                
    Perhaps you meant                                                                                                                                  
      Control.Monad.Trans.Class (from transformers-0.5.6.2)                                                                                            
      Control.Monad.Cont.Class (from mtl-2.2.2)                                                                                                        
      Control.Monad.Error.Class (from mtl-2.2.2)                                                                                                       
    Use -v to see a list of the files searched for.                                                                                                    
   |                                                                                                                                                   
94 | import Control.Comonad.Trans.Class                                                                                                                
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                
                                                                                                                                                       
src/Data/Functor/Foldable.hs:95:1: error:                                                                                                              
    Could not find module ‘Control.Comonad.Trans.Env’                                                                                                  
    Perhaps you meant                                                                                                                                  
      Control.Monad.Trans.Cont (from transformers-0.5.6.2)                                                                                             
      Control.Monad.Trans.RWS (from transformers-0.5.6.2)                                                                                              
    Use -v to see a list of the files searched for.                                                                                                    
   |
95 | import Control.Comonad.Trans.Env
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Data/Functor/Foldable.hs:96:1: error:
    Could not find module ‘Control.Comonad.Cofree’
    Use -v to see a list of the files searched for.
   |
96 | import qualified Control.Comonad.Cofree as Cofree
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Data/Functor/Foldable.hs:97:1: error:
    Could not find module ‘Control.Comonad.Cofree’
    Use -v to see a list of the files searched for.
   |
97 | import Control.Comonad.Cofree (Cofree(..))
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Data/Functor/Foldable.hs:98:1: error:
    Could not find module ‘Control.Comonad.Trans.Cofree’
    Perhaps you meant
      Control.Monad.Trans.Cont (from transformers-0.5.6.2)
    Use -v to see a list of the files searched for.
   |
98 | import           Control.Comonad.Trans.Cofree (CofreeF, CofreeT(..))
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Data/Functor/Foldable.hs:99:1: error:
    Could not find module ‘Control.Comonad.Trans.Cofree’
    Perhaps you meant
      Control.Monad.Trans.Cont (from transformers-0.5.6.2)
    Use -v to see a list of the files searched for.
   |
99 | import qualified Control.Comonad.Trans.Cofree as CCTC
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Data/Functor/Foldable.hs:101:1: error:
    Could not find module ‘Control.Monad.Free’
    Perhaps you meant
      Control.Monad.Fail (from base-4.12.0.0)
      Control.Monad.Fix (from base-4.12.0.0)
      Control.Monad.Cont (from mtl-2.2.2)
    Use -v to see a list of the files searched for.
    |
101 | import Control.Monad.Free (Free(..))
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Data/Functor/Foldable.hs:102:1: error:
    Could not find module ‘Control.Monad.Free.Church’
    Use -v to see a list of the files searched for.
    |
102 | import qualified Control.Monad.Free.Church as CMFC
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Data/Functor/Foldable.hs:104:1: error:
    Could not find module ‘Control.Monad.Trans.Free’
    Perhaps you meant
      Control.Monad.Trans.Cont (from transformers-0.5.6.2)
      Control.Monad.Trans.Error (from transformers-0.5.6.2)
      Control.Monad.Trans.List (from transformers-0.5.6.2)
    Use -v to see a list of the files searched for.
    |
104 | import           Control.Monad.Trans.Free (FreeF, FreeT(..))
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Data/Functor/Foldable.hs:105:1: error:
    Could not find module ‘Control.Monad.Trans.Free’
    Perhaps you meant
      Control.Monad.Trans.Cont (from transformers-0.5.6.2)
      Control.Monad.Trans.Error (from transformers-0.5.6.2)
      Control.Monad.Trans.List (from transformers-0.5.6.2)
    Use -v to see a list of the files searched for.
    |
105 | import qualified Control.Monad.Trans.Free as CMTF
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Makefile:7: recipe for target 'doctest' failed
make: *** [doctest] Error 1
```

My local environment:
```
➜  recursion-schemes git:(master) lsb_release -a
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux 9.9 (stretch)
Release:        9.9
Codename:       stretch
➜  recursion-schemes git:(master) stack --version
Version 2.1.3, Git revision 636e3a759d51127df2b62f90772def126cdf6d1f (7735 commits) x86_64 hpack-0.31.2
➜  recursion-schemes git:(master) cabal --version
cabal-install version 2.4.1.0
compiled using version 2.4.1.0 of the Cabal library
➜  recursion-schemes git:(master) doctest --version
doctest version 0.16.0.1
using version 8.6.5 of the GHC API
using /home/bryan/.stack/programs/x86_64-linux/ghc-8.6.5/bin/ghc-8.6.5
➜  recursion-schemes git:(master) ghc --version
The Glorious Glasgow Haskell Compilation System, version 8.0.1
```

It seems like the `doctest`s are being run successfully in the travis-ci builds, but having looked briefly at the `.travis.yml` file, and at logs from the travis-ci builds, I can't really tell what's going on to get everything set up.

I also haven't found any explicit documentation on how to install, build, and test this project - but I'd be happy to contribute what I can in that area with another PR if I can figure out the proper way to do this myself.

Related question: have we considered using the stack build tool for this project?
In my very limited experience, I've found it relatively easy to work with, but I'm not sure if there are any down sides compared to using cabal directly.

--- a/Makefile
+++ b/Makefile
@@ -1,10 +1,10 @@
 all : build
 
 build :
-	cabal new-build --enable-tests
+	cabal v2-build --enable-tests
 
 doctest :
 	doctest -DCURRENT_PACKAGE_KEY='"recursion-schemes"' -this-unit-id recursion-schemes -Iinclude src
 
 ghcid :
-	ghcid -c 'cabal new-repl'
+	ghcid -c 'cabal v2-repl'

--- a/Makefile
+++ b/Makefile
@@ -1,3 +1,5 @@
+.PHONY: all build doctest ghcid
+
 all : build
 
 build :

--- a/Makefile
+++ b/Makefile
@@ -1,9 +1,14 @@
-.PHONY: all build doctest ghcid
+.PHONY: all build test examples doctest ghcid
 
 all : build
 
 build :
 	cabal v2-build --enable-tests
+
+test : examples doctest
+
+examples :
+	cabal v2-test
 
 doctest :
 	cabal v2-exec -- doctest -DCURRENT_PACKAGE_KEY='"recursion-schemes"' -this-unit-id recursion-schemes -Iinclude src

--- a/Makefile
+++ b/Makefile
@@ -4,7 +4,7 @@ build :
 	cabal v2-build --enable-tests
 
 doctest :
-	doctest -DCURRENT_PACKAGE_KEY='"recursion-schemes"' -this-unit-id recursion-schemes -Iinclude src
+	cabal v2-exec -- doctest -DCURRENT_PACKAGE_KEY='"recursion-schemes"' -this-unit-id recursion-schemes -Iinclude src
 
 ghcid :
 	ghcid -c 'cabal v2-repl'

